relayer/src/link/operational_data.rs

```diff
@@ -17,9 +17,12 @@ use crate::chain::tracking::TrackingId;
 use crate::link::error::LinkError;
 use crate::link::RelayPath;
 
+/// The chain that a piece of `OperationalData` is bound for.
 #[derive(Clone, Copy, PartialEq)]
 pub enum OperationalDataTarget {
+    /// The source chain.
     Source,
+    /// The destination chain.
```

Comment:
```suggestion
    /// The destination chain, i.e., the chain on the receiving side of packets.
```

Reply:
The language being proposed for the `Source` variant vs. the `Destination` variant has me wondering what the relationship is between packets and operational data; how are the two related? Why is it that the destination chain receives packets, not operational data?

Reply:
> what the relationship is between packets and operational data; how are the two related?

This is also something I've been struggling a lot with. Hope these notes below help. The important concepts are marked in __bold__:


- one chain (source) __sends__ packets
    - a user (typically) submits transactions with these kind of packets
- the other chain (destination) __receives__ packets
    - a relayer submits these kind of packets to the destination chain
    - if no relayer acts in time, then the relayer submits a __timeout__ packet to the source chain (instead of **receive** packet to destination chain)
- when a chain processes a send packets, that translates into an event
    - when a chain processes a receive packet, that also translates into an event
    - Hermes acts upon these kind of events reactively
- Hermes translates events into operational data (a concept internal to Hermes), which encompasses both an event as well as any resulting -- receive or timeout -- packet that Hermes has to submit based on that event
- so operational data is just an internal representation that Hermes keeps for pairing together an event with the action that that event entails


aside from the main question, but:
- all packets (send, receive, timeout) are IBC __messages__
- beside packets, there are other types of messages: client update messages, notably, and soon relayer fees messages (I think).
- connection and channel handshakes are also types of IBC messages
- Hermes typically batches multiple IBC messages into a __transaction__

so the three primitive data types are:
- packets, messages, and transactions

I'm thinking we should turn this into a diagram and put in the architecture or docs/ somewhere.

Reply:
I opted to capture as much of this that is relevant to the `OperationalData` type in its doc comment. Hopefully it helps clear some of this up a bit.